File: src/sample/schedule/Section.java
package sample.schedule;

import com.google.gson.annotations.SerializedName;
import javafx.scene.layout.Pane;

public class Section {
    @SerializedName("section_number")
    private String sectionNumber;

    @SerializedName("type")
    private String type;

    @SerializedName("days_of_the_week")
    private String daysOfTheWeek;

    @SerializedName("start_time")
    private int startTime;

    @SerializedName("end_time")
    private int endTime;

    // The transient keyword makes gson not try and parse this.
    // https://stackoverflow.com/questions/4802887/gson-how-to-exclude-specific-fields-from-serialization-without-annotations
    private transient Pane schedulePane;
    private transient boolean hidden = false;

    public static final String[] daysAbbr = {"M", "T", "W", "R", "F"};
    public static final String[] types = {"All", "Lecture", "Discussion", "Laboratory"};

    @Override
    public String toString() {
        return "Section{" +
                "sectionNumber='" + sectionNumber + '\'' +
                ", type='" + type + '\'' +
                ", daysOfTheWeek='" + daysOfTheWeek + '\'' +
                ", startTime=" + startTime +
                ", endTime=" + endTime +
                '}';
    }

    public String getSectionNumber() {
        return sectionNumber;
    }

    public void setSectionNumber(String sectionNumber) {
        this.sectionNumber = sectionNumber;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getDaysOfTheWeek() {
        return daysOfTheWeek;
    }

    public void setDaysOfTheWeek(String daysOfTheWeek) {
        this.daysOfTheWeek = daysOfTheWeek;
    }

    public int getStartTime() {
        return startTime;
    }

    public void setStartTime(int startTime) {
        this.startTime = startTime;
    }

    public int getEndTime() {
        return endTime;
    }

    public void setEndTime(int endTime) {
        this.endTime = endTime;
    }

    public Pane getSchedulePane() {
        return schedulePane;
    }

    public void setSchedulePane(Pane schedulePane) {
        this.schedulePane = schedulePane;
    }

    public void removeSchedulePane() {
        this.schedulePane.setId(this.sectionNumber);
        //remove the pane.
    }

    public boolean hasPane() {
        return this.schedulePane != null;
    }

    public boolean isHidden() {
        return this.hidden;
    }

    public void setHidden(boolean hide) {
        this.hidden = hide;
    }

    public boolean isAtTime(String day, int time) {
        return daysOfTheWeek.contains(day) && time >= startTime && time <= endTime;
    }

    public static String timeIntToString(int value) {
        String output = "";
        int hours = value/100;
        output += (hours < 10 ? '0' : "");
        output += hours;
        output += ":";
        int mins = value%100;
        output += (mins < 10 ? '0' : "");
        output += mins;
        return output;
    }
}

File: CourseScraperPython/Section.py
class Section:
    section_number = None
    type = None
    days_of_the_week = None
    start_time = None
    end_time = None

    def __init__(self, json_dict=None):
        if json_dict is None:
            self.section_number = None
            self.type = None
            self.days_of_the_week = None
            self.start_time = None
            self.end_time = None
        else:
            self.section_number = json_dict['section_number']
            self.type = json_dict['type']
            self.days_of_the_week = json_dict['days_of_the_week']
            self.start_time = json_dict['start_time']
            self.end_time = json_dict['end_time']

    def section_to_string(self):
        output = ""
        output += "SectionNumber: " + self.var_to_string(self.section_number) + "\n"
        output += "Type: " + self.var_to_string(self.type) + "\n"
        output += "Days of the week: " + self.var_to_string(self.days_of_the_week) + "\n"
        output += "Start/End: " + self.var_to_string(self.start_time) \
                  + " to " + self.var_to_string(self.end_time) + "\n"
        return output

    def var_to_string(self, variable):
        if variable is None:
            return "NONE"
        else:
            return str(variable)

    def reprJSON(self):
        return dict(section_number = self.section_number, type=self.type, days_of_the_week = self.days_of_the_week,
                    start_time = self.start_time, end_time = self.end_time)